specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json

```diff
@@ -312,6 +404,33 @@
       },
       "description": "Api output result when disk inspection result is completed."
     },
+    "StorageConfigurationInput": {
+      "type": "object",
+      "properties": {
+        "resourceId": {
+          "description": "Qualified name of the resource.",
+          "type": "string"
+        },
+        "storageAccountId": {
```

Comment:
what is the storage account actually used for and how is it accessed?  What identity is used to get the keys to the storage account if that is the mechanism used to access it? What about storage accounts that have disabled key based auth?

Reply:
1) The new 2 APIs (RegisterStorageConfiguration and ValidateStorageConfiguration) augment the "run" action which executes disk Inspection. These enable a capability where customer need not "remember" which storage account to store results in. StorageAccount information is pinned at subscription level. Details are as follows:
2) The "run" action takes in an "uploadSasUri" parameter as an input in request body. After a successful execution, the results are uploaded to the blob container which is pointed by the input uploadSasUri. This input value is generated outside of RP -> by the caller, either programmatically in Azure Portal using logged in user's identity and accesses or through CLI/PS/SDK where the value is passed in. It can be generated using system key or user-based key as long as it has write permissions. 
2) All the actions call in to /CheckAccess API upfront to validate if caller has correct set of permissions to invoke the underneath use case on the resourceId -> have read permissions on the resourceId to fetch logs from, have read/write permisisons on Storage account to work with.
3) The new APIs (registerStorageConfiguration and ValidateStorageConfiguration) are designed to remember the last known used storage account, similar to cached value so caller need not remember where to have the results uploaded. This will help in E2E Azure Portal experience, where customer need not "remember" or always "select" a storage account from the dropdown control. ValidteStorageConfiguration will return storageaccount id associated at the subscription level, if logged in caller has access to write to it (leverage /CheckAccess API). If not, an error indicating the same is returned through the API. Different calling clients will handle the error differently, either navigate to "Update settings" section or customer call invoke the API through cli by providing a sas that they have access to.

Reply:
gotcha.  What is the "resourceId" in the setConfig action for if the storage account is configured for the entire subscription?

Reply:
Also, what is the "location" URI segment for in registerStorageConfig?  1 config per location per subscription is possible? When validating you choose the same location as the entity that diag is being run against?

Reply:
1) After reviewing the existing functionality, we have agreed to not have "resourceId" information passed in as part of "registerStorageConfiguration" API. The "registerStorageConfiguration" API is intended to create a configuration entry between subscriptionId -> storageAccount (SubscriptionLevel). "ValidateStorageConfiguration" will lookup if such a configuration exists for a subscription and perfom RBAC checks on SA for caller before sending response. I will update swagger and send out another iteration with the change
2) Compute Diagnostic RP is a proxy based, location based routing RP. Hence the path has /location/ for all actions. For Private preview, register action is for 1 config per subscription. Post Public preview, we intend to work off customer feedback to add 1 config per location per subscription changes to have regional affinity of data.